processing/src/test/java/org/apache/druid/segment/TestWrappingDimensionSelector.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.segment;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestWrappingDimensionSelector
+{
+  @Test
+  public void testLongWrappingDimensionSelector()
+  {
+    Long[] vals = new Long[]{24L, null, 50L, 0L, -60L};
+    TestNullableLongColumnSelector lngSelector = new TestNullableLongColumnSelector(vals);
+
+    LongWrappingDimensionSelector lngWrapSelector = new LongWrappingDimensionSelector(lngSelector, null);
+    Assert.assertEquals(24L, lngSelector.getLong());
+    Assert.assertEquals("24", lngWrapSelector.getValue());
+
+    lngSelector.increment();
+    Assert.assertEquals(0L, lngSelector.getLong());
```

Comment:
`lngSelector.getLong` should be never called when the current value is null and `NullHandling.sqlCompatible() == true`. Probably better to check

```java
if (NullHandling.sqlCompatible()) {
  Assert.assertTrue(lngSelector.isNull());
} else {
  Assert.assertEquals(0L, lngSelector.getLong());
}

Reply:
Done, that makes sense.